mmv1/third_party/terraform/website/docs/r/cloudfunctions_function.html.markdown

```diff
@@ -146,7 +146,13 @@ Eg. `"nodejs10"`, `"nodejs12"`, `"nodejs14"`, `"python37"`, `"python38"`, `"pyth
 * `source_archive_object` - (Optional) The source archive object (file) in archive bucket.
 
 * `source_repository` - (Optional) Represents parameters related to source repository where a function is hosted.
-  Cannot be set alongside `source_archive_bucket` or `source_archive_object`. Structure is [documented below](#nested_source_repository).
+  Cannot be set alongside `source_archive_bucket` or `source_archive_object`. Structure is [documented below](#nested_source_repository). It must match the pattern `projects/{project}/locations/{location}/repositories/{repository}`.* 
+
+* `docker_repository` - (Optional, Beta) User managed repository created in Artifact Registry optionally with a customer managed encryption key. If specified, deployments will use Artifact Registry. This is the repository to which the function docker image will be pushed after it is built by Cloud Build.
+
+* `kms_key_name` - (Optional, Beta) Resource name of a KMS crypto key (managed by the user) used to encrypt/decrypt function   
```

Comment:
Since these fields are GA, could you tweak the documentation to say that they're in GA and then specifically mark any beta functionality as beta? For example:

- https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_global_address#purpose
- https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/bigquery_job#source_format

There's not really a standard way to handle this.

Reply:
Updated this. Actually, from the GCF perspective both fields are in the GA. It's just the Terraform support for Artifact Registry that `docker_repository` parameter relies on is still in beta. Clarified that in the latest version.